plasmapy/dispersion/analytical/mhd_wave_class.py

```diff
@@ -0,0 +1,766 @@
+    """
+
+    def angular_frequency(self, k: u.rad / u.m, theta: u.rad):
+        r"""
+        Calculate the angular frequency of magnetohydrodynamic
+        Alfvén waves.
+
+        Parameters
+        ----------
+        k : `~astropy.units.Quantity`, single valued or 1-D array
+            Wavenumber in units convertible to rad/m`.  Either single
+            valued or 1-D array of length :math:`N`.
+        theta : `~astropy.units.Quantity`, single valued or 1-D array
+            The angle of propagation of the wave with respect to the
+            magnetic field, :math:`\cos^{-1}(k_z / k)`, in units must be
+            convertible to radians. Either single valued or 1-D array of
+            size :math:`M`.
+
+        Returns
+        -------
+        omega : `~astropy.units.Quantity`
+            An :math:`N × M` array of computed wave frequencies in units
+            rad/s.
+
+        Raises
+        ------
+        ~astropy.units.UnitTypeError
+            If applicable arguments do not have units convertible to the
+            expected units.
+
+        ValueError
+            If ``k`` is negative or zero.
+
+        ValueError
+            If ``k`` or ``theta`` are not single valued or a 1-D array.
+
+        Warns
+        -----
+        : `~plasmapy.utils.exceptions.PhysicsWarning`
+            When the computed wave frequencies violate the low-frequency
+            (:math:`ω ≪ ω_c,ω_p`) assumption of the dispersion relation.
+
+        Notes
+        -----
+        The angular frequency :math:`\omega` of a magnetohydrodynamic
+        Alfvén wave is given by
+
+        .. math::
+
+            \omega = k v_A \cos\theta
+
+        where :math:`k` is the wavenumber, :math:`v_A` is the Alfvén
+        speed, and :math:`\theta` is the angle between the wavevector and
+        the equilibrium magnetic field.
+
+        Examples
+        --------
+        >>> from astropy import units as u
+        >>> from plasmapy.dispersion.analytical import AlfvenWave
+        >>> alfven = AlfvenWave(1e-3 * u.T, 1e16 * u.m ** -3, "p+")
+        >>> alfven.angular_frequency(1e-5 * u.rad / u.m, 0 * u.deg)
+        <Quantity 2.18060973 rad / s>
+        >>> alfven.angular_frequency([1e-5, 2e-4] * (u.rad / u.m), 0 * u.deg)
+        <Quantity [ 2.18060973, 43.61219459] rad / s>
+        >>> alfven.angular_frequency(1e-5 * u.rad / u.m, [0, 45, 90] * u.deg)
+        <Quantity [2.18060973e+00, 1.54192393e+00, 1.33523836e-16] rad / s>
+        >>> alfven.angular_frequency([1e-5, 2e-4] * (u.rad / u.m), [0, 45, 90] * u.deg)
+        <Quantity [[2.18060973e+00, 1.54192393e+00, 1.33523836e-16],
+                   [4.36121946e+01, 3.08384785e+01, 2.67047673e-15]] rad / s>
+        """
+        theta, k = super()._validate_k_theta(k, theta)
+        omega = k * self._v_a * np.cos(theta)
+        return super()._validate_angular_frequency(omega)
+
+
+class FastMagnetosonicWave(AbstractMHDWave):
+    r"""
+    A class to represent fast magnetosonic waves.
```

Comment:
I wonder if we'd want to add a note somewhere (maybe in the notes section?) that this is also known as the fast magnetoacoustic wave (and similarly for the slow wave below).  That'd help in case people do a search for that term instead.